storage/diemdb/src/lib.rs

```diff
@@ -134,6 +163,24 @@ impl DiemDB {
 
     fn new_with_db(db: DB, prune_window: Option<u64>) -> Self {
         let db = Arc::new(db);
+        let reporting_flag = Arc::new(AtomicBool::new(true));
+        let shared_reporting_flag = Arc::clone(&reporting_flag);
+        let shared_db = Arc::clone(&db);
+
+        let join_handle = Some(thread::spawn(move || loop {
+            if shared_reporting_flag.load(Ordering::Relaxed) == false {
+                break;
+            }
+            let update_result = Self::update_rocksdb_properties(&shared_db);
+            if update_result.is_err() {
+                warn!(
+                    error = ?update_result,
+                    "update rocksdb property failure."
+                );
+            }
+            // report rocksdb properties each 10 second
+            thread::sleep(std::time::Duration::from_secs(10));
```

Comment:
This is potentially hanging the main thread for 10 seconds on exit. Let's use something like `recv_timeout()` on a mpsc receiver.

`while